### PR TITLE
Escape special characters in search by recipient

### PR DESCRIPTION
**Pull request by quis**
SQLAlchemy handles escaping anything that could allow a SQL injection attack. But it doesn’t escape the characters used for wildcard searching. This is the reason we’re able to do `.like('%example%')` at all.

But we shouldn’t be letting our users search with wildcard characters, so we need to escape them. Which is what this commit does.

Documentation: https://www.postgresql.org/docs/9.0/static/functions-matching.html#FUNCTIONS-LIKE

### CODE REVIEW

File: app/dao/notifications_dao.py [hunk 1]
```diff
@@ -455,6 +455,12 @@ def dao_get_notifications_by_to_field(service_id, search_term, notification_type
     else:
         raise InvalidRequest("Only email and SMS can use search by recipient", 400)
 
+    for special_character in {'_', '%', '/'}:
+        normalised = normalised.replace(
+            special_character,
+            '\\{}'.format(special_character)
```
Comment (leohemsted) [inline app/dao/notifications_dao.py]: should this be a raw string? eg `r'\\_'` - or python will convert `\\` into a single backslash

Comment (quis) [reply app/dao/notifications_dao.py]: Single backslash is what I want – maybe would be clearer as `r'\{}'`?

Comment (quis) [reply app/dao/notifications_dao.py]: 9103ca5

Comment (leohemsted) [reply app/dao/notifications_dao.py]: ah i think i got myself confused by reading something from the postgres mailing list from 2000.

`r` is marginally clearer  but not enough for me to mind

**Pull request merged**